Libplanet/Blockchain/BlockChain.cs

```diff
@@ -682,31 +682,51 @@ public long GetNextTxNonce(Address address)
         /// <param name="currentTime">The <see cref="DateTimeOffset"/> when mining started.</param>
         /// <param name="append">Whether to <see cref="Append(Block{T}, StateCompleterSet{T}?)"/>
         /// the mined block.  Turned on by default.</param>
+        /// <param name="txBatchSize">A number to limit the count of transactions that will
+        /// be contained in the block.  <see cref="int.MaxValue"/> by default.</param>
         /// <param name="cancellationToken">
         /// A cancellation token used to propagate notification that this
         /// operation should be canceled.
         /// </param>
         /// <returns>An awaitable task with a <see cref="Block{T}"/> that is mined.</returns>
         /// <exception cref="OperationCanceledException">Thrown when
         /// <see cref="BlockChain{T}.Tip"/> is changed while mining.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when
+        /// <paramref name="txBatchSize"/> is negative.</exception>
         public async Task<Block<T>> MineBlock(
             Address miner,
             DateTimeOffset currentTime,
             bool append = true,
+            int txBatchSize = int.MaxValue,
             CancellationToken cancellationToken = default(CancellationToken)
         )
         {
+            if (txBatchSize < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(txBatchSize));
+            }
+
             long index = Store.CountIndex(Id);
             long difficulty = Policy.GetNextBlockDifficulty(this);
             HashDigest<SHA256>? prevHash = Store.IndexBlockHash(Id, index - 1);
+
             IEnumerable<Transaction<T>> stagedTransactions = Store
                 .IterateStagedTransactionIds()
-                .Select(Store.GetTransaction<T>);
+                .Select(Store.GetTransaction<T>)
+                .GroupBy(tx => tx.Signer)
+                .SelectMany(grp => grp.Select((tx, idx) => (tx, idx)))
+                .OrderBy(t => t.Item2)
+                .Select(t => t.Item1);
 
             var transactionsToMine = new List<Transaction<T>>();
 
             foreach (Transaction<T> tx in stagedTransactions)
             {
+                if (txBatchSize == 0)
+                {
+                    break;
+                }
```

Comment:
How about using [`.Take()`](https://docs.microsoft.com/en-US/dotnet/api/system.linq.enumerable.take?view=netstandard-2.0) instead of checking manually?

Reply:
I applied it in 9618ec1.

Reply:
If we use `.Take()`, insufficient txs will be in the block if some transactions does not follows policy.

Reply:
@limebell's opinion seems right so that I dropped the commit 9618ec1 cc @longfin

Reply:
🆗